src/test/java/org/xerial/snappy/SnappyHadoopCompatibleOutputStreamTest.java

```diff
@@ -0,0 +1,140 @@
+package org.xerial.snappy;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.SystemUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.compress.SnappyCodec;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.*;
+import java.lang.reflect.Field;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class SnappyHadoopCompatibleOutputStreamTest
+{
+
+    private static File tempNativeLibFolder;
+
+    @BeforeClass
+    public static void loadHadoopNativeLibrary() throws Exception
+    {
+        final String libResourceFolder;
+        Map<String, String> libraryNames = new LinkedHashMap<>();
+        if (SystemUtils.IS_OS_LINUX) {
+            libResourceFolder = "/lib/Linux";
+            libraryNames.put("libhadoop.so", "libhadoop.so");
+            // certain Linux systems need these shared library be copied before the JVM started, see build.sbt
+            libraryNames.put("libsnappy.so", "libsnappy.so");
+            libraryNames.put("libsnappy.so.1", "libsnappy.so");
+        } else if (SystemUtils.IS_OS_MAC_OSX) {
+            libResourceFolder = "/lib/MacOSX";
+            libraryNames.put("libhadoop.dylib", "libhadoop.dylib");
+            libraryNames.put("libsnappy.dylib", "libsnappy.dylib");
+            libraryNames.put("libsnappy.1.dylib", "libsnappy.dylib");
+        } else {
```

Comment:
Note: we could do all of these in the sbt, but I feel we should keep the sbt simple. Eventually when the Linux systems have all been upgraded, we could do away those copying in sbt.